internal/server/authn/oidc.go

```diff
@@ -43,6 +45,31 @@ func NewOIDC(domain, clientID, clientSecret, redirectURL string) OIDC {
 	}
 }
 
+// Validate tests if an identity provider has valid attributes to support user login
+func (o *oidcImplementation) Validate() error {
+	ctx := context.Background()
+	conf, _, err := o.clientConfig(ctx)
+	if err != nil {
+		logging.S.Debugf("error validating oidc provider: %s", err)
+		return ErrInvalidProviderURL
+	}
+
+	_, err = conf.Exchange(ctx, "test-code") // 'test-code' is a placeholder for a valid authorization code, it will always fail
+	if err != nil {
+		if strings.Contains(err.Error(), "client_id") || strings.Contains(err.Error(), "client id") {
```

Comment:
I'm wondering if there is any way we could be a little more strict about this. I guess we'll eventually need to test with a bunch of oidc providers to make sure they return the response we expect.

If I'm reading the code correctly, I think the errors we care about from `conf.Exchange` will be https://pkg.go.dev/golang.org/x/oauth2#RetrieveError.

To be a little more strict, we could do something like this:

```go
var errRetrieve *oauth2.RetrieveError
if errors.As(err, &errRetrieve) {
    ...
}
```

That would give us access to both the `http.Request` (to check if the error code is 400), and to the body of the response. Any chance the body of this response is JSON that we could parse? Even if not, checking `strings.Contains(string(errRetrieve.Body)), "client_id")` is a bit more strict.

Reply:
Added this strict error check